### PR TITLE
Wire project addition dialog with project overview page

### PR DESCRIPTION
**Pull request by sven1103**
Introduces the new project collection component and removes the former project overview component.

Project adding is now event-based and orchestrated by its embedded environment.

### CODE REVIEW

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/create/AddProjectDialog.java [hunk 1]
```diff
@@ -310,10 +350,49 @@ private void restrictInputLength() {
...
+    }
+
+    private void configureValidators() {
+      Binder<Container<String>> binderTitle = new Binder<>();
+      binderTitle.forField(titleField)
+          .withValidator(value -> !value.isBlank(), "Please provide a title")
+          .bind(Container::value, Container::setValue);
+      Binder<Container<String>> binderObjective = new Binder<>();
+      binderObjective.forField(projectObjective)
+          .withValidator(value -> !value.isBlank(), "Please provide an " + "objective")
```
Comment (Steffengreiner) [inline vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/create/AddProjectDialog.java:374]: AFAIK we also expect the species specimen and analyte to be set within this dialog and should validate accordingly?

Comment (Steffengreiner) [reply vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/create/AddProjectDialog.java:374]: If that is not the case then the fields should not be set as required

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/create/ProjectAddEvent.java [hunk 2]
```diff
@@ -10,7 +10,7 @@
  *
  * @since 1.0.0
  */
-public class ProjectCreationEvent extends ComponentEvent<ProjectCreationDialog> {
+public class ProjectAddEvent extends ComponentEvent<AddProjectDialog> {
```
Comment (Steffengreiner) [inline vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/create/ProjectAddEvent.java:13]: Please adjust the javadocs accordingly as well :)

**Review by Steffengreiner — CHANGES_REQUESTED** [review]
Heya @sven1103 I found some minor inconsistencies which should be addressed otherwise this PR is good to go

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/create/AddProjectDialog.java (hunk 1 above)
Comment (sven1103) [reply vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/create/AddProjectDialog.java:374]: Hi @Steffengreiner , this is out of scope of this PR. Please open an issue in Jira, if you think this needs to be assessed.

**Review by Steffengreiner — APPROVED** [review]
@sven1103 I'll create a task on our open issue collection to address the missing validation. This PR is g2g

**Pull request merged**